CHANGES.md

```diff
@@ -1,8 +1,461 @@
+  (see TODO and TODO)
+* added support for a new 'Futures' module supporting library-based futures
+  (see TODO)
+* added BLAS level 1 and 2 routines to the previous support for level 3
+  (see TODO)
+* significant performance improvements
+* fixed most memory leaks caused by arrays, domains, and domain maps
+* added new users guide sections on promotion, constants, type aliases, configs
+* added a locale model for KNL with support for different memory types
+  (see TODO)
+* added Chapel support for AWS EC2
+  (see TODO)
+* verified that Chapel works with the Windows 10 bash shell
+  (no platform-specific notes required... works like any other *nix)
+* added support for multi-locale ARM executions
+  (see TODO)
+* added the 'ugni' communication layer to the open-source repository
+* added support for numa-localization of array memory
+
+Deployment
+----------
+* enabled GitHub issues to file and track Chapel bugs and feature requests
+  (see TODO)
+* extended the Docker image to include a GASNet-based configuration
+  (see TODO)
+
+File/Directory Structure
+------------------------
+* moved object files for the compiler and runtime into $CHPL_HOME/build
+* changed the naming/structure of directories used to store binary files
+
+Configuration Changes
+---------------------
+* extended CHPL_LLVM to support new modes: llvm-minimal and system[-minimal]
+  (see TODO)
+
+Semantic Changes / Changes to Chapel Language
+---------------------------------------------
+* made functions return array expressions by value by default
+  (see TODO)
+* made the default intent for arrays be 'ref' if modified, else 'const ref'
+  (see TODO)
+* made the default intent for 'this' on record methods be 'ref' if modified
+  (see TODO)
+* made tuple semantics more closely follow the behavior of their element types
+  (see TODO)
+* deprecated support for the '=>' operator to create array aliases
+  (replace 'var A => B;' with 'ref A = B;' and
+   and 'var A: [D] => B;' with 'ref A = B.reindex(D);')
+* deprecated support for using '=>' in constructors to create alias fields
+
+Syntactic/Naming Changes
+------------------------
+* renamed the 'Error' module to 'SysError'
+  (see TODO)
+
+New Features
+------------
+* added prototype support for error-handling in Chapel
+* dramatically improved support for initializers
+* added support for deinit() as a replacement for class/record destructors
+* added support for 'forwarding' methods to fields of classes and records
+  (see TODO)
+* made 'void' a first-class type in the language and a way to fold variables
+  (see TODO)
+* added support for module deinit() functions
+
+Feature Improvements
+--------------------
+* vastly improved the stability, generality, and precision of 'where' clauses
+  (see TODO)
+* improved array slice, rank change, and reindexing support across domain maps
+* extended return intent overloading to select between 'const' and 'const ref'
```

Comment:
* improved return intent overloading to be useful in more cases
  (e.g. with arrays of arrays)